lib/index.d.ts

```diff
@@ -0,0 +1,286 @@
+            creation: string;
+        }
+        /** Creates a new database. */
+        function create(conn: Connection, database: string, databaseOptions?: DatabaseOptions, options?: Object, params?: Params): Promise<HTTPMessage>;
+        /** Deletes a database. */
+        function drop(conn: Connection, database: string, params?: Params) : Promise<HTTPMessage>;
+        /** Gets information about a database. */
+        function get(conn: Connection, database: string, params?: Params) : Promise<HTTPBody>;
+        /** Sets a database offline. */
+        function offline(conn: Connection, database: string, params?: Params) : Promise<HTTPBody>;
+        /** Sets a database online. */
+        function online(conn: Connection, database: string, params?: Params) : Promise<HTTPBody>;
+        /** Optimizes a database. */
+        function optimize(conn: Connection, database: string, params?: Params) : Promise<HTTPMessage>;
+        /** Makes a copy of a database. */
+        function copy(conn: Connection, database: string, destination: string, params?: Params) : Promise<HTTPBody>;
+        /** Gets a list of all databases on a Stardog server. */
+        function list(conn: Connection, params?: Params) : Promise<HTTPBody>;
+        /** Gets number of triples in a database. */
+        function size(conn: Connection, database: string, databaseOptions?: DatabaseOptions, options?: Object, params?: Params) : Promise<HTTPBody>;
+        /** Clears the contents of a database. */
+        function clear(conn: Connection, database: string, databaseOptions?: DatabaseOptions, options?: Object, params?: Params) : Promise<HTTPBody>;
+        /** Gets a mapping of the namespaces used in a database. */
+        function namespaces(conn: Connection, database: string, databaseOptions?: DatabaseOptions, options?: Object, params?: Params) : Promise<HTTPBody>;
+        /** Exports the contents of a database. */
+        function exportData(conn: Connection, database: string, databaseOptions?: DatabaseOptions, options?: Object, params?: Params) : Promise<HTTPBody>;
+
+        /** Database options. */
+        namespace options {
+            /** Gets set of options on a database. */
+            function get(conn: Connection, database: string, params?: Params) : Promise<HTTPBody>;
+            /** Sets options on a database. */
+            function set(conn: Connection, database: string, databaseOptions: DatabaseOptions, params?: Params) : Promise<HTTPMessage>;
+        }
+
+        interface TransactionOptions extends HTTPOptions {
+            encoding: Encoding
+        }
+        type Encoding =
+            'gzip' |
+            'compress' |
+            'deflate' |
+            'identity' |
+            'br';
+        /** Methods for managing transactions in a database. */
+        namespace transaction {
+            /** Begins a new transaction. */
+            function begin(conn: Connection, database: string, params?: Params) : Promise<HTTPBody>;
```

Comment:
All of the transaction promises also attach `transactionId` to the resolved body to make chaining a little easier.